invenio_files_rest/__init__.py

```diff
@@ -6,7 +6,746 @@
+
+- A robust REST API.
+- Configurable storage backends with the ability to build your very own.
+- Highly customizable permission mechanism for access-control.
+- Secure file handling.
+- Integrity check mechanism.
+- Supports for large file uploads and multipart upload.
+- Signals for system events.
+
+The REST API follows best practices and supports, e.g.:
+
+- Content negotiation and links headers.
+- Cache control via ETags and Last-Modified headers.
+- Optimistic concurrency control via ETags.
+- Rate-limiting, Cross-Origin Resource Sharing, and various security headers.
+
+
+Initialization
+--------------
+First, let's create the flask application:
+
+>>> from flask import Flask
+>>> app = Flask('myapp')
+
+Now to configure our application:
+
+>>> app.config['BROKER_URL'] = 'redis://'
+>>> app.config['CELERY_RESULT_BACKEND'] = 'redis://'
+>>> app.config['DATADIR'] = 'data'
+>>> app.config['FILES_REST_MULTIPART_CHUNKSIZE_MIN'] = 4
+>>> app.config['REST_ENABLE_CORS'] = True
+>>> app.config['SECRET_KEY'] = 'CHANGEME'
+>>> app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite://'
+>>> app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+>>> app.config['FILES_REST_PERMISSION_FACTORY'] = allow_all
+
+
+Now let's initialize all required Invenio extensions:
+
+>>> import shutil
+>>> from os import makedirs
+>>> from os.path import dirname, exists
+>>> from pprint import pprint
+>>> import json
+
+>>> from flask_babelex import Babel
+>>> from flask_menu import Menu
+>>> from invenio_db import InvenioDB, db
+>>> from invenio_rest import InvenioREST
+>>> from invenio_admin import InvenioAdmin
+>>> from invenio_accounts import InvenioAccounts
+>>> from invenio_access import InvenioAccess
+>>> from invenio_accounts.views import blueprint as accounts_blueprint
+>>> from invenio_celery import InvenioCelery
+>>> from invenio_files_rest import InvenioFilesREST
+>>> from invenio_files_rest.views import blueprint
+>>> from invenio_files_rest.models import Location
+
+>>> Babel(app)
+>>> Menu(app)
+>>> InvenioDB(app)
+>>> InvenioREST(app)
+>>> InvenioAdmin(app)
+>>> InvenioAccounts(app)
+>>> InvenioAccess(app)
+
+Finally, let's initialize InvenioFilesREST, register the blueprints
+and push a Flask application context:
+
+>>> InvenioFilesREST(app)
+
+>>> app.register_blueprint(accounts_blueprint)
+>>> app.register_blueprint(blueprint)
+
+>>> app.app_context().push()
+
+Let's create the database and tables, using an in-memory SQLite database:
+
+>>> db.create_all()
+
+>>> srcroot = dirname(dirname(__file__))
+>>> d = app.config['DATADIR']
+>>> if exists(d):
+>>>     shutil.rmtree(d)
+>>> makedirs(d)
+
+There's also a need to create a location for the buckets:
+
+>>> loc = Location(name='local', uri=d, default=True)
+>>> db.session.add(loc)
+>>> db.session.commit()
+
+Now let's create a bucket:
+
+>>> res = app.test_client().post('/files')
+
+And see the response containing the id of the bucket:
+
+>>> pprint(json.loads(res.get_data().decode("utf-8")))
+
+Data Model
```

Comment:
Maybe this clarifications could be treated as the overview in Invenio-Access (see [here](https://invenio-access.readthedocs.io/en/latest/overview.html)), what do you think?